www/js/app.js

```diff
@@ -1017,6 +1051,14 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         // but we should not do this when opening the landing page (or else one of the Unit Tests fails, at least on Chrome 58)
         if (!params.isLandingPage) document.getElementById('articleContent').contentWindow.focus();
 
+        // Early return if article's title is not expected to be displayed
+        // Cause an issue if the current dirEntry is redirected from another dirEntry since the urlExpecedToBedisplayed is not updated
+        // before redirecting.
+        if(! entryIsToExpectedBeDisplayed(dirEntry)){
```

Comment:
It's too early here to do the check, as the content has not been read in the ZIM file yet.
See https://github.com/kiwix/kiwix-js/issues/89#issuecomment-561666315 : it should be done right before displaying the content (at two different lines : one for jQuery mode, one for ServiceWorker mode)